src/utils/resources/vmi/utils/cloud-init.ts

```diff
@@ -1,23 +1,38 @@
+import { load } from 'js-yaml';
+
 import { V1VirtualMachineInstance } from '@kubevirt-ui/kubevirt-api/kubevirt';
 
+type CloudinitUserDataObject = {
+  user?: string;
+  password?: string;
+  passwd?: { users: { name: string }[] };
+};
+
 export const getCloudInitCredentials = (
   vmi: V1VirtualMachineInstance,
-): { user: string; password: string } => {
+): { user?: string; password?: string; users?: string[] } => {
```

Comment:
users?: {name: string}[]

Reply:
Yes. I was not sure if I should convert `user` to array of strings, it would be confusing IMHO. Because we can get user and passowrd, but also list of users from here.. and so we need to deal with that somehow. Or I could rename `user` to `users`? And then simply to return` {users: string[], passord: string}`. WDYT? But I am not sure, it looks confusing that there can be many users but only one password...

Reply:
if you want to only use users , I would suggest doing users: {name: string, password: string | null} , and on rendering, if no password, don't render password field

Reply:
So do you suggest me to rename `user` to `users` and so to return only `users` as the array with objects that contain name and password, right? I can do it so. Just for me personally I'd better go for the way it is now, because it represents the yaml structure/data much better: there can be `user` and `password`.. or `users` - array of objects with `name`, and the rest is optional there.

Reply:
yaml structure is nice to keep if we don't need to do hacks/workarounds on logic. to use only users, not as optional is a much better solution; it prevents us from making login assumptions on .jsx files and makes a complete separation between login and simple rendering files. so yes, much prefer to use only single users[{name: string; password?: string}] :-)

Reply:
we can prevent from this logic on rendering side , const usernames = users || [user || CLOUD_INIT_MISSING_USERNAME]; 

you can also move CLOUD_INIT_MISSING_USERNAME , usage into this function here

Reply:
I think I will leave `CLOUD_INIT_MISSING_USERNAME` in `CloudInitCredentialsContent` component where already is to be able to render "User name: " only when there really is some. Because IMO displaying this would not be nice:
`User name: No credentials, see operating system documentation for the default username. `